packages/awsfirehose/data_stream/metrics/elasticsearch/ingest_pipeline/default.yml

```
@@ -1,9 +1,50 @@
 ---
 description: Pipeline for rerouting metrics stream from Amazon Data Firehose.
 processors:
+  - dot_expander:
+      field: "*"
+      ignore_failure: true
   - set:
       field: ecs.version
       value: 8.11.0
+  - append:
+      field: "aws.metrics_names"
+      value: [ null ]
+      allow_duplicates: false
+      description: Create an empty field of the object type to temporarily store list of all metrics names
+  - script:
+      lang: painless
+      description: This script adds all metric names from 'aws.*.metrics.*.*' into aws.metrics_names field.
+      if: ctx.aws != null
+      source: >-
+        List metricNames = new ArrayList();
+        if (ctx.aws != null && ctx.aws instanceof Map) {
+            for (entry in ctx.aws.entrySet()) {
+                String topLevelKey = entry.getKey();  // Get the top-level key (kinesis, cloudwatch, etc.)
+                def nestedMap = entry.getValue();
+                if (nestedMap instanceof Map) {
+                    for (nestedEntry in nestedMap.entrySet()) {
+                        String key = nestedEntry.getKey(); // Get the second-level key (metrics, namespace, etc.)
+                        if (key.equals("metrics")) {
+                            def metricsMap = nestedEntry.getValue(); // Get aws.*.metrics.*
```

```suggestion
                    def metricsMap = nestedMap.get("metrics"); // Get aws.*.metrics.*
```

I think that should work?